app/src/main/java/io/homeassistant/companion/android/sensors/SensorDetailFragment.kt

```diff
@@ -148,10 +150,13 @@ class SensorDetailFragment(
     private fun updateSensorEntity(
         isEnabled: Boolean
     ) {
-        val sensorEntity = sensorDao.get(basicSensor.id)
+        var sensorEntity = sensorDao.get(basicSensor.id)
         if (sensorEntity != null) {
             sensorEntity.enabled = isEnabled
             sensorDao.update(sensorEntity)
+        } else {
+            sensorEntity = Sensor(basicSensor.id, false, false, "")
```

Comment:
Shouldn't this take into account `isEnabled`?

Reply:
When I was testing it out I tried that but it resulted in the entities being enabled this way they are always kept disabled and they also respect the onboarding permission screen as well.

Reply:
looks like it works as expected now :) thank you for that